File: CS/GridControlWithResizingColumnHeader/Custom GridView/MyGridViewInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Drawing;
using DevExpress.XtraGrid.Drawing;
using DevExpress.Utils;
using DevExpress.Utils.Drawing;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraGrid.Views.Grid;
using System.ComponentModel;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraGrid.Views.Base.ViewInfo;

namespace GridControlWithFindPanel
{
    public class MyGridViewInfo : GridViewInfo
    {
        public MyGridViewInfo(GridView gridView)
            : base(gridView)
        {
        }

        GridHitInfo CalcColumnPanelHitInfo(Point pt, GridHitInfo hi)
        {
            GridColumnInfoArgs ci = CalcColumnHitInfo(pt);
            bool bottomEdge;
            bottomEdge = IntInRangeBottom(pt.Y, ci.Bounds.Bottom, ci.Bounds.Bottom - ControlUtils.ColumnResizeEdgeSize);
            if (bottomEdge)
            {
                hi.HitTest = GridHitTest.RowEdge;
            }
            return hi;
        }

        protected bool IntInRangeBottom(int y, int top, int bottom)
        {
            if (bottom < top)
            {
                int temp = top;
                top = bottom;
                bottom = temp;
            }
            return (y >= top && y < bottom);
        }

        protected internal bool CheckHitTest(Rectangle bounds, Point point, GridHitTest hitTest)
        {
            if (GridDrawing.PtInRect(bounds, point))
            {
                return true;
            }
            return false;
        }

        public override GridHitInfo CalcHitInfo(Point pt)
        {
            GridHitInfo hi = base.CalcHitInfo(pt);
            hi.HitPoint = pt;
            if (!IsReady) return hi;
            if (View.OptionsView.ShowColumnHeaders && CheckHitTest(ViewRects.ColumnPanel, pt, GridHitTest.ColumnPanel))
            {
                return CalcColumnPanelHitInfo(pt, hi);
            }
            return hi;
        }
    }
}

File: VB/GridControlWithResizingColumnHeader/Custom GridView/MyGridViewInfo.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Drawing
Imports DevExpress.XtraGrid.Drawing
Imports DevExpress.Utils
Imports DevExpress.Utils.Drawing
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.XtraGrid.Views.Grid
Imports System.ComponentModel
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.XtraGrid.Views.Base.ViewInfo

Namespace GridControlWithFindPanel
    Public Class MyGridViewInfo
        Inherits GridViewInfo

        Public Sub New(ByVal gridView As GridView)
            MyBase.New(gridView)
        End Sub

        Private Function CalcColumnPanelHitInfo(ByVal pt As Point, ByVal hi As GridHitInfo) As GridHitInfo
            Dim ci As GridColumnInfoArgs = CalcColumnHitInfo(pt)
            Dim bottomEdge As Boolean
            bottomEdge = IntInRangeBottom(pt.Y, ci.Bounds.Bottom, ci.Bounds.Bottom - ControlUtils.ColumnResizeEdgeSize)
            If bottomEdge Then
                hi.HitTest = GridHitTest.RowEdge
            End If
            Return hi
        End Function

        Protected Function IntInRangeBottom(ByVal y As Integer, ByVal top As Integer, ByVal bottom As Integer) As Boolean
            If bottom < top Then
                Dim temp As Integer = top
                top = bottom
                bottom = temp
            End If
            Return (y >= top AndAlso y < bottom)
        End Function

        Protected Friend Function CheckHitTest(ByVal bounds As Rectangle, ByVal point As Point, ByVal hitTest As GridHitTest) As Boolean
            If GridDrawing.PtInRect(bounds, point) Then
                Return True
            End If
            Return False
        End Function

        Public Overrides Function CalcHitInfo(ByVal pt As Point, Optional ByVal ignoreData As Boolean = False) As GridHitInfo
            Dim hi As GridHitInfo = MyBase.CalcHitInfo(pt)
            hi.HitPoint = pt
            If Not IsReady Then
                Return hi
            End If
            If View.OptionsView.ShowColumnHeaders AndAlso CheckHitTest(ViewRects.ColumnPanel, pt, GridHitTest.ColumnPanel) Then
                Return CalcColumnPanelHitInfo(pt, hi)
            End If
            Return hi
        End Function
    End Class
End Namespace
